src/client/containers/export-data/export-data.component.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import Button from '../../components/button/button.component';
+// import PropTypes from 'prop-types';
+
+const ExportData = () => {
+  const objectTocsv = function(data) {
+    const csvRows = [];
+
+    // get the headers
+    const headers = Object.keys(data[0]);
+    csvRows.push(headers.join(','));
+
+    // loop over the rows & format the data
+    data.forEach((row) => {
+      const values = headers.map((header) => {
+        const escaped = `${row[header]}`.replace(/"/g, '\\"');
+        return `"${escaped}"`;
+      });
+      csvRows.push(values.join(','));
+    });
+    return csvRows.join('\n');
+  };
+
+  //   Method to download csv file from browser - Browsers that support HTML5 download attribute
+  const download = function(data) {
+    const blob = new Blob([data], { type: 'text/csv' });
+    const url = window.URL.createObjectURL(blob);
+    const a = document.createElement('a');
+    a.setAttribute('hidden', '');
+    a.setAttribute('href', url);
+    a.setAttribute('download', 'exportedData.csv');
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+  };
+
+  //   Fetching data from endpoint
+  const getRequest = async function() {
+    const jsonUrl = 'https://jsonplaceholder.typicode.com/users';
```

Comment:
Can you make the jsonUrl a prop? it will be easier to share with other graphs.

Reply:
Hi Ricardo, Thank you for your suggestion - I ll look into that. But one other issue linked with code is, It works when I use external URL link 'https://jsonplaceholder.typicode.com/users' but I does not work on sensor reading endpoint #150  'http://localhost:3000/api/sensor-reading/' .
Component is lying in client/containers/export-data.
May its because of authentication / wrong folder placement or something else?

Reply:
There is good examples of consuming API running at your localhost https://dzone.com/articles/consuming-rest-api-with-reactjs